core/node/base_token_adjuster/src/base_token_ratio_persister.rs

```diff
@@ -33,6 +40,7 @@ pub struct BaseTokenRatioPersister {
     base_token_address: Address,
     price_api_client: Arc<dyn PriceAPIClient>,
     l1_params: Option<BaseTokenRatioPersisterL1Params>,
+    last_persisted_l1_ratio: Arc<RwLock<Option<BigDecimal>>>,
```

Comment:
hmm perhaps hairsplitting but alternatively you could request the ratio on component initialization - then the behaviour wouldn't depend on component restart (I think now on restart you'd always reset the value on the smart contract).

In that case this value would only be a `None` when `l1_params` above is `None`. 

Actually, I'm not a big fan of this `l1_params: Option<...>` - which is `Some` or `None` depending on the config passed - and then somewhere inside the implementation we dynamically check its value. Alternatively we could do something like `enum BaseTokenL1Behaviour {UpdateOnL1, NoOp} ` and implement all the l1-related logic in its `impl`. 

I won't block the merge by this, but please think about it and do this change if you like it.

Reply:
I wouldn't make any assumptions about number of restarts if possible

Reply:
Ok 2vs1 - deal, will read the prev ratio from the chain on startup.

Reply:
Addressed both points - fetching ratios from L1 on start and moved L1 logic to a separate struct

Reply:
thanks! re: setting value on restart - I agree it's not very important, but for me it's not so much spending gas money, but more it doesn't feel right that the L1 behaviour depends on restarts. imagine someone looking at the list of updates - they always update when the price differs by 10%, but _sometimes_ just in random times.

Thanks for addressing!